krun/config.py

```diff
@@ -41,16 +41,13 @@ def _fatal_exception_execing_config(self, exc_info):
             lines.append(frame)
         fatal("".join(lines))
 
-    def read_from_string(self, config_str):
-        config_dict = {}
-        try:
-            exec(config_str, config_dict)
-        except Exception as e:
-            self._fatal_exception_execing_config(sys.exc_info())
-
-        self.__dict__.update(config_dict)
-        self.filename = ""
-        self.text = config_str
+    def check_config_consistency(self, config_str):
+        import difflib
+        if self.text != config_str:
+            diff = "\n".join(difflib.unified_diff(self.text.splitlines(),
+                                                  config_str.splitlines()))
+            fatal("the config file %s on disk was different than "
+                  "the resumed config file:\n%s" % (self.filename, diff))
```

Comment:
Sorry, I find "mid-experiment" less precise. How about "the config file ... is different than the stored copy in ..._result.json.bz2"